### PR TITLE
Demzne 1145 factor out flexible data fields for tables into reusable class

### PR DESCRIPTION
**Pull request by angelol**
Use new SINGLETON macro

### CODE REVIEW

**Comment by dskvr** [comment]
Task linked: [DEMZNE-1145 Factor out flexible data fields for tables into reusable class](https://app.clickup.com/t/18330827/DEMZNE-1145)

File: contract-shared-headers/daccustodian_shared.hpp [hunk 1]
```diff
@@ -256,75 +225,16 @@ namespace eosdac {
...
-        PROPERTY(uint32_t, initial_vote_quorum_percent);
-        PROPERTY(uint32_t, vote_quorum_percent);
-        PROPERTY(uint8_t, auth_threshold_high);
-        PROPERTY(uint8_t, auth_threshold_mid);
-        PROPERTY(uint8_t, auth_threshold_low);
-        PROPERTY(uint32_t, lockup_release_time_delay);
-        PROPERTY(eosio::extended_asset, requested_pay_max);
-        PROPERTY(uint64_t, token_supply_theshold);
-    };
+    SINGLETON(dacglobals, daccustodian, PROPERTY_OPTIONAL_TYPECASTING(uint16_t, uint32_t, budget_percentage);
```
Comment (dallasjohnson) [inline contract-shared-headers/daccustodian_shared.hpp:228]: I assume this is probably a clangformat limitation but can this list each PROPERTY on a new line to help with readability? I love this feature cleanup. Very elegant 👏

File: contracts/daccustodian/config.cpp [hunk 2]
```diff
@@ -61,22 +61,19 @@ ACTION daccustodian::updateconfige(const contr_config &new_config, const name &d
...
     print("Succesfully updated the daccustodian config for: ", dac_id);
 }
 
 ACTION daccustodian::setbudget(const name &dac_id, const uint16_t percentage) {
     require_auth(get_self());
 
-    auto globals = dacglobals::current(get_self(), dac_id);
+    auto globals = dacglobals{get_self(), dac_id};
     globals.set_budget_percentage(percentage);
-    globals.save(get_self(), dac_id);
```
Comment (dallasjohnson) [inline contracts/daccustodian/config.cpp:73]: Good to see these `save` calls can be safely removed due to the destructor. This was an easy bug to run into.

**Review by dallasjohnson — APPROVED** [review]
LGTM 👍

File: contract-shared-headers/daccustodian_shared.hpp (hunk 1 above)
Comment (angelol) [reply contract-shared-headers/daccustodian_shared.hpp]: I just remembered you can turn off clang-format for sections of code. It looks much better now!

**Review by dallasjohnson — APPROVED** [review]
LGTM 👍

**Pull request merged**